### PR TITLE
Bomtool changes

### PR DESCRIPTION
**Pull request by bamandel**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -114,7 +114,7 @@ license {
     ext.year = Calendar.getInstance().get(Calendar.YEAR)
     ignoreFailures = true
     includes (["**/*.groovy", "**/*.java"])
-    exclude "/src/test/*"
+    excludes (["/src/test/*.groovy", "src/test/*.java"])
```
Comment (jrichardBD) [inline build.gradle]: Why was this necessary to add?

Comment (bamandel) [reply build.gradle]: It wasn't correctly excluding the test files previously as far as I was aware. If we had to specifically include the .java and .groovy extensions in the includes field, it would make sense to do the same for excludes.

File: src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/NpmBomTool.groovy [hunk 2]
```diff
@@ -95,9 +97,16 @@ class NpmBomTool extends BomTool {
...
-                logger.warn("Could not find a ${executableManager.getExecutableName(ExecutableType.NPM)} executable")
+                logger.warn("Could not find an ${executableManager.getExecutableName(ExecutableType.NPM)} executable")
             } else {
-                logger.debug("Npm version ${executableRunner.runExe(npmExePath, '-version').standardOutput}")
+                npmLsExe = new Executable(new File(sourcePath), npmExePath, ['-version'])
+                String npmNodePath = detectConfiguration.getNpmNodePath()
+                int lastSlashIndex = npmNodePath.lastIndexOf('/')
+                npmNodePath = npmNodePath.substring(0, lastSlashIndex)
+                if (npmNodePath) {
+                    npmLsExe.environmentVariables.put('PATH', npmNodePath)
```
Comment (jrichardBD) [inline src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/NpmBomTool.groovy]: Is Node the only executable that NPM requires to be available on the PATH?

Comment (bamandel) [reply src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/NpmBomTool.groovy]: In eclipse I replaced the path with an almost empty one and using these properties, npm ran just fine.
